components/admin_console/team_channel_settings/team/details/team_profile.test.tsx

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import configureStore from 'redux-mock-store';
+import * as reactRedux from 'react-redux';
+import {shallow} from 'enzyme';
+
+import {mountWithIntl} from 'tests/helpers/intl-test-helper';
+
+import {CloudProducts} from 'utils/constants';
+import {FileSizes} from 'utils/file_utils';
+
+import {TeamProfile} from './team_profile.jsx';
+
+describe('admin_console/team_channel_settings/team/TeamProfile__Cloud', () => {
+    const baseProps = {
+        team: {
+            display_name: 'test',
+        },
+        onToggleArchive: jest.fn(),
+        isArchived: true,
+    };
+    const initialState = {
+        views: {
+            announcementBar: {
+                announcementBarState: {
+                    announcementBarCount: 1,
+                },
+            },
+        },
+        entities: {
+            general: {
+                config: {
+                    FeatureFlagCloudFree: 'true',
+                } as any,
+                license: {
+                    IsLicensed: 'true',
+                    Cloud: 'true',
```

Comment:
Can we add a test to verify that Upgrade is not enabled in non-cloud?

Reply:
There is one here: https://github.com/mattermost/mattermost-webapp/pull/10387/files#diff-a2f743bfcdfe3e3bede45fa1b2ef85ccefe3e816648426efb24409e3b129206bR315

The reason the test works properly and the spinwick doesn't is because the usage is zero'd in the test, which is how I expected it to work for non-cloud contexts

Reply:
Gotcha, thanks!